scipy/signal/ltisys.py

```diff
@@ -1106,3 +1112,453 @@ def freqresp(system, w=None, n=10000):
+    
+    method: string (default "YT")        
+        Which method to choose to find K.
+        "KNV0": Kautsky et Al. Robust pole assignment in linear state feedback,
+            Int journal of Control 1985, vol 41 p 1129->1155)        
+        "YT" (Tits, Yang. Globally Convergent Algorithms for Robust Pole
+            Assignment by State Feedback http://drum.lib.umd.edu/handle/1903/5598). 
+    
+    maxtry: integer, optionnal (default 20)
+        Maximum number of iterations to compute K.
+        
+    force_matry: boolean, optional (default False)
+        By default K will be returned as soon as an iteration does not yield
+        any significant change on the determinant of the eigenvalues of A-BK.
+        If force_maxtry is set to True K will be returned after maxtry steps.
+        
+
+    return_poles: boolean, optional (default False)
+        Whether to return or not the actual placed poles.
+        
+    Returns
+    -------
+    K : 1D ndarray
+        The closed loop matrix such as the eigenvalues of A-BK are as close
+        as possible to the requested poles P.
+    Pa : 1D ndarray (optional)
+        The poles corresponding to A-BK.
+
+    Example
+    --------
+    #Test real pole placement using KNV and YT algorithm
+    #Example number 1 from section 4 of the reference KNV publication
+    
+    A=np.array([1.380,-0.2077,6.715,-5.676, -0.5814,-4.290,0,0.6750,
+            1.067,4.273,-6.654,5.893, 0.0480,4.273,1.343,-2.104]).reshape(4,4)
+    B=np.array([0,5.679,1.136,1.136,0,0,-3.146,0]).reshape(4,2)
+    P=np.array((-0.2,-0.5,-5.0566,-8.6659))
+        
+    #compute K with KNV method 0
+    K1,P1=place(A,B,P, method="KNV0",return_poles=True)
+    #compute K with YT    
+    K2,P2=place(A,B,P, return_poles=True)
+    #note K1 and K2 may differ as the solution is not unique
+
+
+    """
+    #©Irvin Probst, Ensta Bretagne, 2014
+    #changelog:
+    #2/12/2014 
+    #-initial post in scipy-dev ML
+    #3/12/2014
+    #-when B is square we don't need to run the optimisation
+    #algorithms as there is only one solution => linalg.solve
+    #-added normalization of Xj vectors 
+    #-added Yang Tits optimisation of KNV method 0
+    #-removed useless KNV method 1
+    #4/12/2014
+    #-use bette variable names (S->Ker)
+    #5/12/2014
+    #-finished Yang Tits
+    #-code clean up
+    #-improved convergence check on X to stop the main loop earlier if
+    #  we have found a solution (is that really useful ?)
+    #8/12/2014 
+    #-added complex support for YT, deduced how to make it work for KNV0 too
+    #9/12/2014
+    #-fixed *again* convergence test
+    #10->12/12/2014
+    #-bugfixes, bugfixes, bugfixes...
+    #-removed complex support from KNV0 as it is not reliable enough 
+    #17/12/2014
+    #-improved transfer matrix initial selection
+
+    #play it safe safe with whatever the users might pass as P
+    P = sort(array(P).flatten())[::-1]  
```

Comment:
Is it ok to do something like:

    poles=array(poles)        
    if len(poles.shape)>1:
        raise ValueError("Poles must be a 1D array like.") 

And rely on array() to raise ValueError if the user inputs some dumb poles such as (1,(2,(3))) ? Or shall I catch this ValueError and raise some more explicit message ?